### DISCUSSION TITLE
Is it possible any DropIn Worflow sample?

### DISCUSSION THREAD

**Discussion by ericwu-glife** [elsa-workflows/elsa-core#5058] [Q&A]
There is a sample project for the DropIns activities and it's working fine.
Is there any sample for the DropIns workflows?
Thanks.

**Comment by glime-ai** [comment]
As of the information available up to the knowledge cutoff date, there is no specific mention of a sample project for DropIns workflows in the provided context. The references to DropIns primarily discuss activities and the ability to load them dynamically. However, if you are looking for examples of how to use workflows with DropIns, you might consider looking at the general documentation and samples for Elsa workflows and attempt to adapt those examples to work with the DropIns feature.

For more detailed guidance or examples, you may want to check the official Elsa documentation, GitHub repository, or reach out to the community on the Elsa Discord channel for any unofficial samples or community-contributed resources that may exist.

If you need further assistance or specific guidance on implementing DropIns workflows, please provide additional details about your requirements or the issues you are facing.

**Comment by raymonddenhaan** [comment]
Hey @ericwu-glife, could you elaborate what you mean with a DropIn activity/workflow?

**Reply by ericwu-glife** [reply]
The sample project provide the way to add activities:

- Before MsDI: `IDropIn.Install()`
- After MsDI: `IDropIn.ConfigureAsync()` with `IServiceProvider`

I'm expecting to find the way to add workflows by the IDropIn before/after MsDI.
I have implemented to load JSON workflows on `IDropIn.ConfigureAsync()` only.
My question is to expecting the way to load CLR workflows with the `IDropIn`.
```csharp
public class DropIn : IDropIn
{
    public void Install(IModule module)
    {
        module.AddActivitiesFrom<DropIn>();
    }

    public async ValueTask ConfigureAsync(IServiceProvider serviceProvider, CancellationToken cancellationToken)
    {
        var activitReg = serviceProvider.GetRequiredService<IActivityRegistry>();
        await activitReg.RegisterAsync<DumpImg>(cancellationToken);

        var serializer = serviceProvider.GetRequiredService<IActivitySerializer>();
        var importer = serviceProvider.GetRequiredService<IWorkflowDefinitionImporter>();

        var asm = typeof(DropIn).Assembly;
        var resNames = asm.GetManifestResourceNames()
            .Where(n => n.EndsWith(".json") && n.Contains(".Workflows."));
        foreach(var resName in resNames)
        {
            using(var stream = asm.GetManifestResourceStream(resName))
            using(var reader = new StreamReader(stream!)) 
            {
                var json = reader.ReadToEnd();
                var wfModel = serializer.Deserialize<WorkflowDefinitionModel>(json);
                var varNamesCount = wfModel.Variables?
                    .Select(v => v.Name)?.ToArray()?
                    .Intersect(VariableDefinitionEx.NAMES_Required)?.Count();
                if (varNamesCount is null || varNamesCount.Value == 0) continue;
                
                var wfRequest = new SaveWorkflowDefinitionRequest
                {
                    Model = wfModel
                };
                await importer.ImportAsync(wfRequest);
            }
        }
    }
}
```

**Comment by glime-ai** [comment]
Based on the information provided, it seems that there is no direct mention of a sample project specifically for DropIns workflows in the context given. However, there is a reference to a sample project for DropIns activities which is working fine. If you are looking for examples of DropIns workflows, you might want to explore the sample project for DropIns activities as it could potentially include workflow examples or at least give you an idea of how to implement workflows using DropIns.

Additionally, you could check the Elsa GitHub repository for any examples or reach out to the community on the Elsa Discord channel or GitHub issues for further assistance. Since the documentation and samples are evolving, it's possible that more examples will be added in the future, so keeping an eye on the official Elsa documentation and repository updates would be beneficial.

If you need further clarification or assistance, please let me know what specific aspects of DropIns workflows you are interested in, and I can provide more targeted information.
